### PR TITLE
migrate to sbt 1.0.2

### PR DESCRIPTION
**Pull request by larsrh**
Gets rid of this funny "compile-time" hack because sbt 1.0.x complains about something.

I have it on good notice that "provided" should be Good Enough™.

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -57,16 +58,8 @@ val commonSettings = Seq(
     }
   },
 
-  // credit: https://github.com/typelevel/cats/pull/1638
-  ivyConfigurations += config("compile-time").hide,
-  unmanagedClasspath in Compile ++= update.value.select(configurationFilter("compile-time")),
-
   logBuffered in Test := false,
 
-  headers := Map(
```
Comment (ChristopherDavenport) [inline build.sbt]: Where did these go on the upgraded versions of sbt-header?

Comment (larsrh) [reply build.sbt]: This is auto-detected from other settings now.

**Comment by dwijnand** [comment]
I think provided will keep it in pom.xml while compile-time won't. I think.

**Comment by alexandru** [comment]
@dwijnand isn't `compile-time` the default, aka the strongest dependency possible?

AFAIK when you want to exclude something from the `pom.xml`, you have to do a `pomPostProcess` — not sure how well it works in SBT 1.x

**Comment by dwijnand** [comment]
What do you mean by "the default, aka the strongest dependency possible"?

It's an ad-hoc configuration created in this build (and cats).

It's an alternative to doing `pomPostProcess`, with, I think, the added value that it also works for ivy.xml (if publishing ivy style).

**Comment by larsrh** [comment]
It is true that `macro-compat` is still in the POM, but clients won't be pulling in the library, because it's in the `provided` scope. If this is a contentious issue, I can just change it back.

**Comment by BennyHill** [comment]
> If this is a contentious issue

Well, I think it was in cats - https://github.com/typelevel/cats/pull/1638, https://github.com/sbt/sbt/issues/2503 

perhaps this would help https://github.com/typelevel/cats/pull/1889 ?

**Comment by larsrh** [comment]
I have reinstated the configuration now.

**Pull request merged**